### PR TITLE
Update to Guava 28.1

### PR DESCRIPTION
**Pull request by elharo**

### CODE REVIEW

**Comment by ejona86** [comment]
The transitive version checks are failing, and repositories.bzl also needs to be updated.

**Comment by elharo** [comment]
This appears to run into a conflict with auth:


```
/usr/local/google/home/elharo/grpc-java/auth/src/test/java/io/grpc/auth/GoogleAuthLibraryCallCredentialsTest.java:327: error: constructor ServiceAccountCredentials in class ServiceAccountCredentials cannot be applied to given types;
    ServiceAccountCredentials credentials = new ServiceAccountCredentials(
                                            ^
  required: String,String,PrivateKey,String,Collection<String>,HttpTransportFactory,URI,String,String
  found: <null>,String,PrivateKey,<null>,<null>
  reason: actual and formal argument lists differ in length
/usr/local/google/home/elharo/grpc-java/auth/src/test/java/io/grpc/auth/GoogleAuthLibraryCallCredentialsTest.java:354: error: constructor ServiceAccountCredentials in class ServiceAccountCredentials cannot be applied to given types;
    ServiceAccountCredentials credentials = new ServiceAccountCredentials(
                                            ^
  required: String,String,PrivateKey,String,Collection<String>,HttpTransportFactory,URI,String,String
  found: <null>,String,PrivateKey,<null>,List<String>
  reason: actual and formal argument l
```

That is upgrading auth to the first version that is compatible with j2objc 1.3, which is version 0.16.0, introduces an API incompatibility.

**Comment by ejona86** [comment]
@elandau, those failures are only in tests. There is a builder that can be used instead.

File: auth/src/test/java/io/grpc/auth/GoogleAuthLibraryCallCredentialsTest.java [hunk 1]
```diff
@@ -350,15 +350,17 @@ public AccessToken refreshAccessToken() {
   public void serviceAccountWithScopeNotToJwt() throws Exception {
```
[the remark this answers is not in the document]
Comment (elharo) [reply auth/src/test/java/io/grpc/auth/GoogleAuthLibraryCallCredentialsTest.java]: done

File: auth/src/test/java/io/grpc/auth/GoogleAuthLibraryCallCredentialsTest.java [hunk 2]
```diff
@@ -323,14 +325,12 @@ public void serviceUri() throws Exception {
   @Test
   public void serviceAccountToJwt() throws Exception {
     KeyPair pair = KeyPairGenerator.getInstance("RSA").generateKeyPair();
-    @SuppressWarnings("deprecation")
-    ServiceAccountCredentials credentials = new ServiceAccountCredentials(
```
[the remark this answers is not in the document]
Comment (elharo) [reply auth/src/test/java/io/grpc/auth/GoogleAuthLibraryCallCredentialsTest.java]: I'll try this. It may take a little while.

**Pull request merged**